core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -1139,32 +1277,78 @@ private void executeInsert() {
                     break;
             }
             prepareCommandComplete(true);
-        } catch (CairoException e) {
+        } catch (Throwable e) {
             if (transactionState == IN_TRANSACTION) {
                 transactionState = ERROR_TRANSACTION;
             }
             throw e;
         }
     }
 
-    private CharSequence getStatement0(long lo, long hi) throws BadProtocolException {
-        CharacterStoreEntry e = characterStore.newEntry();
-        if (Chars.utf8Decode(lo, hi, e)) {
-            return characterStore.toImmutable();
-        } else {
-            LOG.error().$("invalid UTF8 bytes in statement name").$();
-            throw BadProtocolException.INSTANCE;
+    private void executeTag() {
+        LOG.debug().$("executing [tag=").$(queryTag).$(']').$();
+        if (queryTag != null && TAG_OK != queryTag) {  //do not run this for OK tag (i.e.: create table)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.